test/python/pulse/test_schedule.py

```diff
@@ -518,31 +564,77 @@ def test_filter_multiple(self):
+                                                              instruction_types=[PulseInstruction],
+                                                              time_ranges=[(25, 100), (0, 11)])
+        self.assertTrue(len(excluded.instructions), 3)
+        for time, inst in filtered.instructions:
             self.assertIsInstance(inst, (FrameChangeInstruction, PulseInstruction))
-        self.assertTrue(len(filtered_b.instructions), 4)
+        self.assertTrue(len(filtered.instructions), 4)
+        # make sure the PulseInstruction not in the intervals is maintained
+        self.assertIsInstance(excluded.instructions[0][1], PulseInstruction)
 
-    def test_filter(self):
-        """Test _filter method."""
+    def test_custom_filters(self):
+        """Test custom filters."""
         device = self.two_qubit_device
         lp0 = self.linear(duration=3, slope=0.2, intercept=0.1)
         sched = Schedule(name='fake_experiment')
         sched = sched.insert(0, lp0(device.drives[0]))
         sched = sched.insert(10, lp0(device.drives[1]))
         sched = sched.insert(30, FrameChange(phase=-1.57)(device.drives[0]))
-        for i in sched._filter([lambda x: True]).instructions:
+
+        filtered, excluded = self.filter_and_test_consistency(sched, [lambda x: True])
+        for i in filtered.instructions:
             self.assertTrue(i in sched.instructions)
-        self.assertEqual(len(sched._filter([lambda x: False]).instructions), 0)
-        self.assertEqual(len(sched._filter([lambda x: x[0] < 30]).instructions), 2)
+        for i in excluded.instructions:
+            self.assertFalse(i in sched.instructions)
+
+        filtered, excluded = self.filter_and_test_consistency(sched, [lambda x: False])
+        self.assertEqual(len(filtered.instructions), 0)
+        self.assertEqual(len(excluded.instructions), 3)
+
+        filtered, excluded = self.filter_and_test_consistency(sched, [lambda x: x[0] < 30])
+        self.assertEqual(len(filtered.instructions), 2)
+        self.assertEqual(len(excluded.instructions), 1)
+
+        # multiple custom filters
+        filtered, excluded = self.filter_and_test_consistency(sched, [lambda x: x[0] > 0,
+                                                                      lambda x: x[0] < 30])
+        self.assertEqual(len(filtered.instructions), 1)
+        self.assertEqual(len(excluded.instructions), 2)
+
+        # multiple custom filters specified as unkeyed arguments
+        filtered, excluded = self.filter_and_test_consistency(sched,
+                                                              lambda x: x[0] > 0,
+                                                              lambda x: x[0] < 30)
+        self.assertEqual(len(filtered.instructions), 1)
+        self.assertEqual(len(excluded.instructions), 2)
+
+    def filter_and_test_consistency(self, schedule, *args, **kwargs):
+        """
+        This is a helper function which returns the tuple
```

Comment:
This should be written in a "command" or "recipe" form: `Return the tuple... and assert that ....`